subworkflows/local/align_star/nextflow.config

```diff
@@ -1,20 +1,32 @@
 if (!params.skip_alignment && params.aligner == 'star_salmon') {
     process {
         withName: '.*:ALIGN_STAR:STAR_ALIGN|.*:ALIGN_STAR:STAR_ALIGN_IGENOMES' {
-            ext.args   = { [
-                '--quantMode TranscriptomeSAM',
-                '--twopassMode Basic',
-                '--outSAMtype BAM Unsorted',
-                '--readFilesCommand zcat',
-                '--runRNGseed 0',
-                '--outFilterMultimapNmax 20',
-                '--alignSJDBoverhangMin 1',
-                '--outSAMattributes NH HI AS NM MD',
-                '--quantTranscriptomeBan Singleend',
-                '--outSAMstrandField intronMotif',
-                params.save_unaligned ? '--outReadsUnmapped Fastx' : '',
-                params.extra_star_align_args ? params.extra_star_align_args.split("\\s(?=--)") : ''
-            ].flatten().unique(false).join(' ').trim() }
+            ext.args   = {
```

Comment:
Principle makes sense.

Would it make more sense to factor out maps and merge from there? Like the following (untested). 

```
ext.args = {
    // Function to convert argument strings into a map
    def argsToMap(String args) {
        args.split("\\s(?=--)").collectEntries {
            def (key, value) = it.trim().split(/\s+/, 2)
            [(key): value]
        }
    }

    // Initialize the map with preset arguments
    def preset_args_map = argsToMap('''
        --quantMode TranscriptomeSAM
        --twopassMode Basic
        --outSAMtype BAM Unsorted
        --readFilesCommand zcat
        --runRNGseed 0
        --outFilterMultimapNmax 20
        --alignSJDBoverhangMin 1
        --outSAMattributes NH HI AS NM MD
        --quantTranscriptomeBan Singleend
        --outSAMstrandField intronMotif
        ${params.save_unaligned ? '--outReadsUnmapped Fastx' : ''}
    '''.trim())

    // Convert extra arguments into a map
    def extra_args_map = params.extra_star_align_args ? argsToMap(params.extra_star_align_args) : [:]

    // Merge maps: extra arguments override preset arguments
    def final_args_map = preset_args_map + extra_args_map

    // Convert the map back to a list and then to a single string
    final_args_map.collect { key, value -> "${key} ${value}" }.join(' ').trim()
}

```

Reply:
Either way is fine with me. 

Since I do know neither Java nor Groovy (_why is Nextflow not pythonic :-/ ?!?_) I was just happy that I could nudge the AI into coming up with something that worked. But it is true, using a map is an elegant solution, because it avoids repeatedly looping over the preset arguments for each extra argument.